google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java

```diff
@@ -0,0 +1,374 @@
+                conditions.add(new ConditionV4(type, "cache-control", cacheControl));
+                return this;
+            }
+
+            public Builder addContentDispositionCondition(
+                    ConditionV4Type type, String contentDisposition) {
+                conditions.add(new ConditionV4(type, "content-disposition", contentDisposition));
+                return this;
+            }
+
+            public Builder addContentEncodingCondition(ConditionV4Type type, String contentEncoding) {
+                conditions.add(new ConditionV4(type, "content-encoding", contentEncoding));
+                return this;
+            }
+
+            public Builder addContentLengthCondition(ConditionV4Type type, int contentLength) {
+                conditions.add(new ConditionV4(type, "content-length", "" + contentLength));
+                return this;
+            }
+
+            public Builder addContentTypeCondition(ConditionV4Type type, String contentType) {
+                conditions.add(new ConditionV4(type, "content-type", contentType));
+                return this;
+            }
+
+            public Builder addExpiresCondition(ConditionV4Type type, long expires) {
+                conditions.add(new ConditionV4(type, "expires", dateFormat.format(expires)));
+                return this;
+            }
+
+            public Builder addExpiresCondition(ConditionV4Type type, String expires) {
+                conditions.add(new ConditionV4(type, "expires", expires));
+                return this;
+            }
+
+            public Builder addKeyCondition(ConditionV4Type type, String key) {
+                conditions.add(new ConditionV4(type, "key", key));
+                return this;
+            }
+
+            public Builder addSuccessActionRedirectUrlCondition(
+                    ConditionV4Type type, String successActionRedirectUrl) {
+                conditions.add(
+                        new ConditionV4(type, "success_action_redirect", successActionRedirectUrl));
+                return this;
+            }
+
+            public Builder addSuccessActionStatusCondition(ConditionV4Type type, int status) {
+                conditions.add(new ConditionV4(type, "success_action_status", "" + status));
+                return this;
+            }
+
+            public Builder addCustomMetadataCondition(ConditionV4Type type, String field, String value) {
+                conditions.add(new ConditionV4(type, "x-goog-meta-" + field, value));
+                return this;
+            }
+
+            public Builder addContentLengthRangeCondition(int min, int max) {
+                conditions.add(
+                        new ConditionV4(ConditionV4Type.CONTENT_LENGTH_RANGE, "" + min, "" + max));
+                return this;
+            }
+
+            public Builder addCustomCondition(ConditionV4Type type, String field, String value) {
```

Comment:
Suggest removing this until we get a feature request for this method.

Reply:
We actually use it in the generate method, but I can change it to package private

Reply:
package private sgtm, thanks for clarifying!